src/sentry/grouping/ingest.py

```diff
@@ -295,3 +298,84 @@ def find_existing_grouphash(
             )
 
     return None, root_hierarchical_hash
+
+
+def get_hash_values(
+    project: Project,
+    job: Job,
+    metric_tags: MutableTags,
+) -> tuple[CalculatedHashes, CalculatedHashes | None, CalculatedHashes]:
+    # Background grouping is a way for us to get performance metrics for a new
+    # config without having it actually affect on how events are grouped. It runs
+    # either before or after the main grouping logic, depending on the option value.
+    _maybe_run_background_grouping(project, job)
+
+    secondary_hashes = None
+
+    if _should_run_secondary_grouping(project):
+        with metrics.timer("event_manager.secondary_grouping", tags=metric_tags):
+            secondary_grouping_config = SecondaryGroupingConfigLoader().get_config_dict(project)
+            secondary_hashes = _calculate_secondary_hash(project, job, secondary_grouping_config)
+
+    with metrics.timer("event_manager.load_grouping_config"):
+        # At this point we want to normalize the in_app values in case the
+        # clients did not set this appropriately so far.
+        if is_reprocessed_event(job["data"]):
+            # The customer might have changed grouping enhancements since
+            # the event was ingested -> make sure we get the fresh one for reprocessing.
+            grouping_config = get_grouping_config_dict_for_project(project)
+            # Write back grouping config because it might have changed since the
+            # event was ingested.
+            # NOTE: We could do this unconditionally (regardless of `is_processed`).
+            job["data"]["grouping_config"] = grouping_config
+        else:
+            grouping_config = get_grouping_config_dict_for_event_data(
+                job["event"].data.data, project
+            )
```

Comment:
FWIW, it's totally beyond scope of this PR, but I have strong suspicion that this whole section is leftover cruft that could be removed.

Reply:
I'm not 100% sure, as I'll admit I haven't spent a ton of time digging into the reprocessing logic. Agreed it's out of scope here, but I may well need to come back to this part of the code later on, so let's table it for now but keep it in mind for the future.